migrations/versions/088f00b10226_.py

```diff
@@ -0,0 +1,139 @@
+"""
+1.2.0 - Add biospecimen_genomic_file association table
+
+Revision ID: 088f00b10226
+Revises: ce359a457d3d
+Create Date: 2018-07-12 07:41:56.857583
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql import table
+from sqlalchemy.dialects import postgresql
+import dataservice
+from datetime import datetime
+from dataservice.api.common.id_service import uuid_generator, kf_id_generator
+
+# revision identifiers, used by Alembic.
+revision = '088f00b10226'
+down_revision = 'ce359a457d3d'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('biospecimen_genomic_file',
+                    sa.Column('uuid', postgresql.UUID(), nullable=True),
+                    sa.Column('created_at', sa.DateTime(), nullable=True),
+                    sa.Column('modified_at', sa.DateTime(), nullable=True),
+                    sa.Column('genomic_file_id', dataservice.api.common.model.KfId(
+                        length=11), nullable=False),
+                    sa.Column('biospecimen_id', dataservice.api.common.model.KfId(
+                        length=11), nullable=False),
+                    sa.Column('kf_id', dataservice.api.common.model.KfId(
+                        length=11), nullable=False),
+                    sa.ForeignKeyConstraint(['biospecimen_id'], [
+                                            'biospecimen.kf_id'], ),
+                    sa.ForeignKeyConstraint(['genomic_file_id'], [
+                                            'genomic_file.kf_id'], ),
+                    sa.PrimaryKeyConstraint('kf_id'),
+                    sa.UniqueConstraint('genomic_file_id', 'biospecimen_id'),
+                    sa.UniqueConstraint('uuid')
+                    )
+    data_upgrades()
+    op.drop_column('genomic_file', 'biospecimen_id')
+
+
+def data_upgrades():
+    """
+    COPY genomic_file.biospecimen_id TO biospecimen_genomic_file.biospecimen_id
+    COPY genomic_file.kf_id TO biospecimen_genomic_file.genomic_file_id
+    DROP genomic_file.biospecimen_id
+    """
+    connection = op.get_bind()
+
+    biospecimen_genomic_file = sa.Table(
```

Comment:
These table instantiations look like we're creating the tables again or something..I see a lot of duplicate code here and above in upgrade(). Is there a way to get a reference to `biospecimen_genomic_file` table and `genomic_file` table without defining all of the columns?

Reply:
We need to define these manually because using the code defined in the models can be unreliable during an upgrade Where the tables in the database could be any sort of schema.